fuel-vm/src/tests/coins.rs

```diff
@@ -0,0 +1,884 @@
+    ];
+    let script_data: Vec<u8> = [Call::new(this_contract, 0, 0).to_bytes().as_slice()]
+        .into_iter()
+        .flatten()
+        .copied()
+        .chain(this_contract.to_bytes())
+        .chain(asset_id.to_bytes())
+        .collect();
+
+    let result = test_context
+        .start_script(script_ops, script_data)
+        .script_gas_limit(1_000_000)
+        .contract_input(this_contract)
+        .fee_input()
+        .contract_output(&this_contract)
+        .execute();
+
+    extract_novalue(result.receipts())
+}
+
+#[test_case(false, 0, 0, 10 => RunResult::Panic(PanicReason::TransferZeroCoins); "(external) Cannot transfer 0 coins to non-Variable output")]
+#[test_case(false, 1, 0, 10 => RunResult::Panic(PanicReason::TransferZeroCoins); "(external) Cannot transfer 0 coins to valid output")]
+#[test_case(false, 9, 0, 10 => RunResult::Panic(PanicReason::TransferZeroCoins); "(external) Cannot transfer 0 coins to non-existing output")]
+#[test_case(false, 1, 1, 10 => RunResult::Success((1, 9)); "(external) Can transfer 1 coins")]
+#[test_case(false, 1, 11, 10 => RunResult::Panic(PanicReason::NotEnoughBalance); "(external) Cannot transfer just over balance coins")]
+#[test_case(false, 1, Word::MAX, 0 => RunResult::Panic(PanicReason::NotEnoughBalance); "(external) Cannot transfer max over balance coins")]
+#[test_case(false, 1, Word::MAX, Word::MAX => RunResult::Success((Word::MAX, 0)); "(external) Can transfer Word::MAX coins")]
+#[test_case(false, 0, 1, 10 => RunResult::Panic(PanicReason::OutputNotFound); "(external) Target output is not Variable")]
+#[test_case(false, 9, 1, 1 => RunResult::Panic(PanicReason::OutputNotFound); "(external) Target output doesn't exist")]
+#[test_case(true, 0, 0, 10 => RunResult::Panic(PanicReason::TransferZeroCoins); "(internal) Cannot transfer 0 coins to non-Variable output")]
+#[test_case(true, 1, 0, 10 => RunResult::Panic(PanicReason::TransferZeroCoins); "(internal) Cannot transfer 0 coins to valid output")]
+#[test_case(true, 9, 0, 10 => RunResult::Panic(PanicReason::TransferZeroCoins); "(internal) Cannot transfer 0 coins to non-existing output")]
+#[test_case(true, 1, 1, 10 => RunResult::Success((1, 9)); "(internal) Can transfer 1 coins")]
+#[test_case(true, 1, 11, 10 => RunResult::Panic(PanicReason::NotEnoughBalance); "(internal) Cannot transfer just over balance coins")]
+#[test_case(true, 1, Word::MAX, 0 => RunResult::Panic(PanicReason::NotEnoughBalance); "(internal) Cannot transfer max over balance coins")]
+#[test_case(true, 1, Word::MAX, Word::MAX => RunResult::Success((Word::MAX, 0)); "(internal) Can transfer Word::MAX coins")]
+#[test_case(true, 0, 1, 10 => RunResult::Panic(PanicReason::OutputNotFound); "(internal) Target output is not Variable")]
+#[test_case(true, 9, 1, 1 => RunResult::Panic(PanicReason::OutputNotFound); "(internal) Target output doesn't exist")]
+fn transfer_to_output(
+    internal: bool,
+    to_index: Word, // 1 = the variable output
```

Comment:
I know that it should affect anything, but could you also test with `u64::MAX` please?=)

Reply:
Sure. 1e93295c